src/Event/Http/Psr7Bridge.php

```diff
@@ -23,22 +23,33 @@ final class Psr7Bridge
      */
     public static function convertRequest(HttpRequestEvent $event, Context $context): ServerRequestInterface
     {
+        $headers = $event->getHeaders();
+
         [$files, $parsedBody] = self::parseBodyAndUploadedFiles($event);
+        [$user, $password] = self::parseBasicAuthorization($headers);
 
-        $server = [
-            'SERVER_PROTOCOL' => $event->getProtocolVersion(),
+        $server = array_filter([
+            'CONTENT_LENGTH' => $headers['content-length'][0] ?? null,
+            'CONTENT_TYPE' => $event->getContentType(),
+            'DOCUMENT_ROOT' => getcwd(),
+            'QUERY_STRING' => $event->getQueryString(),
             'REQUEST_METHOD' => $event->getMethod(),
```

Comment:
I'd say it wouldn't hurt to also add:

```php
'SERVER_ADDR' => '127.0.0.1',
'SCRIPT_FILENAME' => $_SERVER['_HANDLER'] ?? $_SERVER['SCRIPT_FILENAME'],
'GATEWAY_INTERFACE' => 'FastCGI/1.0',
```

Reply:
My way of thinking was that these imply FastCGI and/or a local reverse proxy, but that doesn't have to be the case since this is just a PSR-7 bridge.

Reply:
Agreed, specifically these might be overkill and even misleading?